### ISSUE TITLE
RadioButton React - Screen reader cannot read label

### ISSUE THREAD

**Issue by keithkade** [carbon-design-system/carbon#4491]
## Environment

> Operating system
OS X
> Browser
Chrome
> Assistive technology used to verify
VoiceOver / JAWS

## Detailed description

> What version of the Carbon Design System are you using?
10 
> What did you expect to happen?
The screen reader can read individual radio button content
> What happened instead?
It can't find it

## Steps to reproduce the issue

1. Activate VoiceOver
2. Select a radio button in the vanilla html (it reads the content)
2. Select a radio button in the React version (it doesn't)

This can be recreated on the main component documentation https://www.carbondesignsystem.com/components/radio-button/code and the react storybook page http://react.carbondesignsystem.com/?path=/story/radiobuttongroup--default

## Additional information

Here's what is reads in the vanilla component. The important bit is that is correctly gets "Radio button label"
![image](https://user-images.githubusercontent.com/2426829/67813459-9505a600-fa6f-11e9-92de-14895f0d76a7.png)

Here's what it reads in the react component. It can't read the label
![image](https://user-images.githubusercontent.com/2426829/67813465-97680000-fa6f-11e9-8d5d-2c394805560e.png)

I dug into the code a bit here's the vanilla html:
```
<div class="bx--radio-button-wrapper">
   <input id="radio-button-unp8hnypvgd-1" class="bx--radio-button" type="radio" value="red" name="radio-button--vertical" tabindex="0" checked>
   <label for="radio-button-unp8hnypvgd-1" class="bx--radio-button__label">
   <span class="bx--radio-button__appearance"></span>
   <span class="bx--radio-button__label-text">Radio button label</span>
   </label>
</div>
```
and here is the html generated from the react component
```
<div class="some-class bx--radio-button-wrapper">
   <input id="radio-1" name="radio-button-group" type="radio" class="bx--radio-button" value="standard">
   <label for="radio-1" class="bx--radio-button__label" aria-label="Radio button label">
   <span class="bx--radio-button__appearance"></span>
   <span class="">Radio button label</span>
   </label>
</div>
```

I think the issue is the aria-label="Radio button label"` in the react version, which isn't needed.

**Comment by snidersd** [comment]
@dakahn  I just retested this on mac w/Chrome & VO and confirmed it fails. 
I also tested it on MacOS mac with Safari and in this case the label is announced twice. 
and last,  when tested on Win10 with JAWS on Chrome or FF, as well as iOS with Safari and VO it works as expected.

**Comment by carmacleod** [comment]
Good find, @keithkade. You are correct that the `aria-label="Radio button label"` isn't needed, because it's already in the content of the label.

@dakahn The `aria-label` attribute on the `<label>` should just be deleted. That will fix the VO+Safari duplication. Not sure if it will fix the VO+Chrome problem (needs testing), but it is incorrect code that should be deleted anyhow.
